include/s3select_functions.h

```diff
@@ -643,6 +646,23 @@ struct _fn_substr : public base_function{
     
 };
 
+struct _fn_lower : public base_function {
+
+    value v_str;
+    char buff[4096];
```

Comment:
I think please just use std::string here?

Reply:
it's normally not good style to put a large buffer on the stack, and in c++, we try to avoid direct char buffer manipulation in general

Reply:
we need to discuss that.
this object is actually on heap, same for the buff.
this object and such (operator implementations) are allocate by *new* , in this case placement-new.
declaring buff on functor, allocate it only once, and not on stuck.
with std::string , it may call ::new (global new), using std::string is correct(flexible,reusable) , but reviewing perf , it may have side affect.